src/zarr/core/array.py

```diff
@@ -603,9 +609,24 @@ async def getitem(
         )
         return await self._get_selection(indexer, prototype=prototype)
 
-    async def _save_metadata(self, metadata: ArrayMetadata) -> None:
+    async def _save_metadata(self, metadata: ArrayMetadata, ensure_parents: bool = False) -> None:
```

Comment:
The new keyword is to ensure that updates to an existing nodes don't require all the `setdefault` operations to ensure that parents exist. Anytime we create a brand new node we should call `_save_metdata` with `ensure_parents=True`.